File: guides/micronaut-scope-types/java/src/main/java/example/micronaut/infrastructure/Robot.java
package example.micronaut.infrastructure;

import io.micronaut.context.annotation.Infrastructure;
import io.micronaut.core.annotation.NonNull;

import java.util.UUID;

@Infrastructure
public class Robot {
    @NonNull
    private final String serialNumber;

    public Robot() {
        serialNumber = UUID.randomUUID().toString();
    }

    @NonNull
    public String getSerialNumber() {
        return serialNumber;
    }
}

File: guides/micronaut-scope-types/python/src/example/micronaut/refreshable/robot.py
from uuid import uuid4

from micronaut.runtime.context.scope import Refreshable


@Refreshable  # <1>
class Robot:
    def __init__(self):
        self._serial_number = str(uuid4())

    @property
    def serial_number(self) -> str:
        return self._serial_number
